libs/agno/agno/eval/criteria.py

```diff
@@ -0,0 +1,819 @@
+        console.print(results_table)
+
+
+@dataclass
+class CriteriaResult:
+    """Aggregated results from criteria evaluations."""
+
+    results: List[CriteriaEvaluation] = field(default_factory=list)
+    avg_score: float = field(init=False)
+    min_score: float = field(init=False)
+    max_score: float = field(init=False)
+    std_dev_score: float = field(init=False)
+    pass_rate: float = field(init=False)
+
+    def __post_init__(self):
+        self.compute_stats()
+
+    def compute_stats(self):
+        import statistics
+
+        if self.results and len(self.results) > 0:
+            scores = [r.score for r in self.results]
+            passed = [r.passed for r in self.results]
+            self.avg_score = statistics.mean(scores)
+            self.min_score = min(scores)
+            self.max_score = max(scores)
+            self.std_dev_score = statistics.stdev(scores) if len(scores) > 1 else 0.0
+            self.pass_rate = sum(passed) / len(passed) * 100
+        else:
+            self.avg_score = 0.0
+            self.min_score = 0.0
+            self.max_score = 0.0
+            self.std_dev_score = 0.0
+            self.pass_rate = 0.0
+
+    def print_summary(self, console: Optional["Console"] = None):
+        from rich.box import ROUNDED
+        from rich.console import Console
+        from rich.table import Table
+
+        if console is None:
+            console = Console()
+
+        summary_table = Table(
+            box=ROUNDED,
+            border_style="blue",
+            show_header=False,
+            title="[ Criteria Evaluation Summary ]",
+            title_style="bold sky_blue1",
+            title_justify="center",
+        )
+        summary_table.add_row("Number of Evaluations", f"{len(self.results)}")
+        summary_table.add_row("Pass Rate", f"{self.pass_rate:.1f}%")
+        summary_table.add_row("Average Score", f"{self.avg_score:.2f}/10")
+        summary_table.add_row("Min Score", f"{self.min_score:.2f}/10")
+        summary_table.add_row("Max Score", f"{self.max_score:.2f}/10")
+        if self.std_dev_score > 0:
+            summary_table.add_row("Std Deviation", f"{self.std_dev_score:.2f}")
+        console.print(summary_table)
+
+    def print_results(self, console: Optional["Console"] = None):
+        for result in self.results:
+            result.print_eval(console)
+
+
+@dataclass
+class CriteriaEval(BaseEval):
+    """Evaluate agent outputs using custom criteria with an LLM judge."""
+
+    # Core evaluation fields
+    criteria: str = ""
+    threshold: int = 7
+    on_fail: Optional[Callable[["CriteriaEvaluation"], None]] = None
+    additional_guidelines: Optional[Union[str, List[str]]] = None
+    additional_context: Optional[str] = None
+
+    # Evaluation metadata
+    name: Optional[str] = None
+    eval_id: str = field(default_factory=lambda: str(uuid4()))
+    num_iterations: int = 1
+    run_id: Optional[str] = None
+    result: Optional[CriteriaResult] = None
```

Comment:
This would be stateful to store a result. It should just be returned. Same with run_id